app/models/health_care_application.rb

```diff
@@ -66,17 +66,25 @@ def process!
     end
   end
 
-  def self.enrollment_status(icn)
+  def self.enrollment_status(icn, loa3)
     ee_data = HCA::EnrollmentEligibility::Service.new.lookup_user(icn)
-    parsed_status = HCA::EnrollmentEligibility::StatusMatcher.parse(
-      ee_data[:enrollment_status], ee_data[:ineligibility_reason]
-    )
 
-    ee_data.slice(
-      :application_date,
-      :enrollment_date,
-      :preferred_facility
-    ).merge(parsed_status: parsed_status)
+    if loa3
+      parsed_status = HCA::EnrollmentEligibility::StatusMatcher.parse(
+        ee_data[:enrollment_status], ee_data[:ineligibility_reason]
+      )
+
+      ee_data.slice(
+        :application_date,
+        :enrollment_date,
+        :preferred_facility
+      ).merge(parsed_status: parsed_status)
```

Comment:
Would there be any value abstracting this logic out into a private method, named after its business purpose?  I'm not sure what `ee_data` is (I should probably dig in further), but something like:

```ruby
eligibility_data_for(x, y)
```
Or something to that effect?  

Perhaps even a new instance method on the `HCA::EnrollmentEligibility::Service` that you could call on `ee_data`?  Just some thoughts, you'd know better.

Reply:
sorry, i accidentally started working on another issue in this ticket, i'll keep your comments in mind for the other issue